### ISSUE TITLE
🚀 Feature: Ability to discard a Change

### ISSUE THREAD

**Issue by phlrss** [novuhq/novu#3311]
### 🔖 Feature description

It would be greatly beneficial if we had the ability to "Discard" a Change rather than only "Promote" it. If we make changes to a template in Development, and then want to fine-tune or even revert the change there is no way to "Discard" it so it remains in the Change table indefinitely.

### 🎤 Why is this feature needed ?

Without it, old no longer relevant changes will remain trailing in the Change table unnecessarily. 

### ✌️ How do you aim to achieve this?

Simply add a "Discard" button beside the "Promote" button with a dialog "Are you sure you want to discard this change?" to avoid any accidental discards. And perform a delete by ID in the `changes` table in the database

### 👀 Have you spent some time to check if this feature request has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Contributing Guidelines](https://github.com/novuhq/novu/blob/main/CONTRIBUTING.md)

### Are you willing to submit PR?

None

**Comment by cedric-fauth** [comment]
Also it would be great to improve the changes check to see if there are really any changes made or if nothing has changed.
When calling the api and updating a workflow twice with the same data there are 2 changes displayed even if nothing has changed since the first update. You should consider checking for changes based on actual data changes inside the workflows and displaying a diff so that everyone knows what has changed.

**Comment by scopsy** [comment]
@gitstart posting my Pr comment here as well:
```
 As It might affect some other underlying changes once the entity will be removed. We will need to do some deeper investigation before we can go and implement something like this. My suggestion is to close this for the time being until we finalize a proper flow for this.
 ```

One of the challenges is how to deal with the already made changes on dev if we discard the change from the list, subsequent save can just return the unexpected change as a diff is calculated for each time a workflow is updated

**Comment by bartelemi** [comment]
To add even more context to this issue, there is one more situation where this feature would be helpful. Currently we got stuck in a situation where we made some changes to a workflow that was subsequently removed, but the "change" was never promoted.

Trying to promote such change just errors with a message: "_Notification Group Id 111a11dbb1a1aa111a1a1111 not found, Notification Template: Mention in a comment_" (ID redacted) and is kind of stuck in a limbo.

**Comment by scopsy** [comment]
We are moving towards publish and sync based approach in v2, so closing this ticket as it not something we are going to support
